### ISSUE TITLE
Jedi Academy become very lag after installed OpenJK

### ISSUE THREAD

**Issue by mfon223** [JACoders/OpenJK#1065]
before i install openjk, everything works fine. After i installed OpenJK, the game just gone very lag that cause me cant even move on the menu, but when i use windows 7 it work fine, after upgraded to windows 10, everytime time i install openjk the game gone very low fps like 1, i dont know is my opengl problem or something else, can somebody help me fix this issue??

**Comment by ensiform** [comment]
We need to know more information about your system like the graphics card and drivers if any and seeing log files. If it's a laptop that is useful information too.

You can generate logfiles by starting the game by adding the following launch parameters to a shortcut or batch file.

`+set logfile 2`

So a shortcut target field would look like for example:

`"D:\Games\JKA\GameData\openjk_sp.x86.exe" +set logfile 2`

The actual path on your machine will differ.

**Comment by mfon223** [comment]
Device name	user-PC
Processor	Intel(R) Core(TM) i5-2540M CPU @ 2.60GHz   2.60 GHz
Installed RAM	4.00 GB (3.88 GB usable)
Device ID	726FF58A-0715-402B-A114-631DFECAEC6D
Product ID	00330-50000-00000-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	Touch support with 2 touch points


Edition	Windows 10 Pro
Version	20H2
Installed on	‎9/‎7/‎2020
OS build	19042.630
Experience	Windows Feature Experience Pack 120.2212.31.0

**Comment by ensiform** [comment]
Judging from that it seems like you have a laptop and it is probably running off of your integrated graphics.

**Comment by mfon223** [comment]
yes i use laptop, so is there any way to fix the lag on openjk after my laptop upgraded to windows 10? originally is windows 7

**Comment by ensiform** [comment]
I gave instructions on how to get a logfile that can be more useful in diagnosing your problem otherwise there is nothing we can do to help.

**Comment by mfon223** [comment]
it didnt work

**Comment by xycaleth** [comment]
Here's a few things to try out. Go through them one by one:

1. Make sure your graphics drivers are up to date. I don't see any mention of a dedicated graphics chip so I'm guessing you have an integrated one. You can download the Intel graphics drivers for your CPU here: https://downloadcenter.intel.com/product/50072/Intel-Core-i5-2540M-Processor-3M-Cache-up-to-3-30-GHz- and choose "Intel® Graphics Driver for Windows* [15.28]"
2. In `My Documents\My Games\OpenJK\base`, there should be a file called `openjk.cfg`. Try renaming this to `openjk.old.cfg` and start OpenJK again to see if that fixes the FPS issue. It might be that some existing settings don't play nicely with your upgraded Windows 10 installation.
3. If none of the above works, then we need a bit more information. Load up OpenJK, bring down the console by pressing the key to the left of the <kbd>1</kbd> key and then entering `/condump output.log` and press Enter. This will write the contents of the console to `My Documents\My Games\OpenJK\output.log`. Copy/paste the contents of that file here. This won't fix anything but it'll help us diagnose the problem a bit easier.

**Comment by mfon223** [comment]
![intel](https://user-images.githubusercontent.com/66712674/102883778-87895780-448b-11eb-8d57-3f60d2baab95.PNG
should i ignore not or dont install?

**Comment by mfon223** [comment]
OpenJK-MP: v1.0.1.0 win_msvc-x86 Dec 21 2020
----- FS_Startup -----
Current search path:
C:\Users\user\Documents\My Games\OpenJK\base
C:\GOG Games\Star Wars Jedi Knight - Jedi Academy\GameData\base\assets3.pk3
(16 files)
C:\GOG Games\Star Wars Jedi Knight - Jedi Academy\GameData\base\assets2.pk3
(62 files)
C:\GOG Games\Star Wars Jedi Knight - Jedi Academy\GameData\base\assets1.pk3
(8320 files)
C:\GOG Games\Star Wars Jedi Knight - Jedi Academy\GameData\base\assets0.pk3
(15346 files)
C:\GOG Games\Star Wars Jedi Knight - Jedi Academy\GameData\base

----------------------
23744 files in pk3 files
execing mpdefault.cfg
execing openjk.cfg
couldn't exec autoexec.cfg
----- Initializing Renderer ----
Trying to load "rd-vanilla_x86.dll" from "C:\GOG Games\Star Wars Jedi Knight
- Jedi Academy\GameData"...
QKEY found.
SDL using driver "windows"
Initializing display
Display aspect: 1.778
...setting mode 4: 800 600
Using 24 color bits, 24 depth, 8 stencil display.
Available modes: '1280x720 1366x768 1600x900 1360x768 1280x768 320x200
640x400 768x480 960x600 1280x800 1440x900 320x240 400x300 512x384 640x480
800x600 1024x768'
GL_RENDERER: GDI Generic
Initializing OpenGL extensions
...no supported texture compression method found
.....ignoring texture compression
...GL_EXT_texture_env_add not found
...GL_EXT_texture_filter_anisotropic not found
...using GL_EXT_texture_edge_clamp
...GL_ARB_multitexture not found
...GL_EXT_compiled_vertex_array not found
...GL_NV_register_combiners not found
...GL_ARB_vertex_program not found
...GL_ARB_fragment_program not found

GL_VENDOR: Microsoft Corporation
GL_RENDERER: GDI Generic
GL_VERSION: 1.1.0
GL_WIN_swap_hint GL_EXT_bgra GL_EXT_paletted_texture
GL_MAX_TEXTURE_SIZE: 1024
GL_MAX_TEXTURE_UNITS_ARB: 0

PIXELFORMAT: color(24-bits) Z(24-bit) stencil(8-bits)
MODE: 4, 800 x 600 windowed hz:N/A
GAMMA: hardware w/ 0 overbright bits
rendering primitives: multiple glArrayElement
texturemode: GL_LINEAR_MIPMAP_NEAREST
picmip: 0
texture bits: 0
multitexture: disabled
compiled vertex arrays: disabled
texenv add: disabled
compressed textures: disabled
compressed lightmaps: disabled
texture compression method: None
anisotropic filtering: disabled  Dynamic Glow: disabled

------- sound initialization -------
SDL_Init( SDL_INIT_AUDIO )... OK
SDL audio driver is "wasapi".
SDL_AudioSpec:
  Format:   AUDIO_S16LSB
  Freq:     44100
  Samples:  1024
  Channels: 2
Starting SDL audio callback...
SDL audio initialized.
----- Sound Info -----
    1 stereo
32768 samples
   16 samplebits
    1 submission_chunk
44100 speed
0x4635838 dma buffer
No background file.
----------------------
------------------------------------

--- ambient sound initialization ---
Sound memory manager started
VM_Create: uix86.dll failed!
VM_CreateLegacy: uix86.dll succeeded
--- Common Initialization Complete ---
SDL Version Compiled: 2.0.12
SDL Version Linked: 2.0.12
Winsock Initialized
Hostname: user-PC
IP: 192.168.1.2
Opening IP socket: localhost:29070
]condump output.log
Con_Dump_f: Only the ".txt" extension is supported by this command!
]condump output.txt
Dumped console text to output.txt.

**Comment by xycaleth** [comment]
Here’s the problem!
```
GL_VENDOR: Microsoft Corporation
GL_RENDERER: GDI Generic
GL_VERSION: 1.1.0
```
The game thinks OpenGL isn’t available so it’s falling back to the software renderer.

> ![intel](https://user-images.githubusercontent.com/66712674/102883778-87895780-448b-11eb-8d57-3f60d2baab95.PNG
> should i ignore not or dont install?

Can you check Windows Updates to see if you have any updates pending? If not then try installing the drivers I linked

**Comment by mfon223** [comment]
> Here’s the problem!
> 
> ```
> GL_VENDOR: Microsoft Corporation
> GL_RENDERER: GDI Generic
> GL_VERSION: 1.1.0
> ```
> 
> The game thinks OpenGL isn’t available so it’s falling back to the software renderer.
> 
> > ![intel](https://user-images.githubusercontent.com/66712674/102883778-87895780-448b-11eb-8d57-3f60d2baab95.PNG
> > should i ignore not or dont install?
> 
> Can you check Windows Updates to see if you have any updates pending? If not then try installing the drivers I linked

Yes, Cumulative Update for Windows 10 Version 20H2 for x64-based Systems (KB4592438), this updates pending only, no others

**Comment by kittycat67** [comment]
Hello guys im having the same issue with open JK really low FPS

**Comment by UnknownNumber2** [comment]
So i have the same problem altough i started playing on windows 10. I do have a laptop and the first thing i tried is fixing the game settings because i thought my laptop got bad but the fps stayed the same with high quality and low quality. I did the thing in the files where you add the .old but also didn't work. What can i do?

**Comment by UnknownNumber2** [comment]
Yeah updating windows didn't help at all.

**Comment by UnknownNumber2** [comment]
> > Here’s the problem!
> > ```
> > GL_VENDOR: Microsoft Corporation
> > GL_RENDERER: GDI Generic
> > GL_VERSION: 1.1.0
> > ```
> > 
> > 
> > The game thinks OpenGL isn’t available so it’s falling back to the software renderer.
> > > ![intel](https://user-images.githubusercontent.com/66712674/102883778-87895780-448b-11eb-8d57-3f60d2baab95.PNG
> > > should i ignore not or dont install?
> > 
> > 
> > Can you check Windows Updates to see if you have any updates pending? If not then try installing the drivers I linked
> 
> Yes, Cumulative Update for Windows 10 Version 20H2 for x64-based Systems (KB4592438), this updates pending only, no others

It says the exact thing on my console altough i did all the updates.

**Comment by UnknownNumber2** [comment]
![Στιγμιότυπο οθόνης (69)](https://user-images.githubusercontent.com/81254917/112205770-d75bf400-8c1d-11eb-96c2-62969541c68c.png)

**Comment by Razish** [comment]
Can we close this issue as "user hardware/driver problem"?
I don't see this as an issue with OpenJK necessarily, but with particular GPU drivers. All the above examples are using the software renderer as your drivers are not providing OpenGL support. That's an issue with your driver, and likely due to jamp.exe only being supported by a hacky fix inside your driver.

If someone is still affected by this _and_ the original/retail engine works fine, can you try renaming OpenJK's executable to `jamp.exe` and tell us if that works?
Then can you try renaming the original `jamp.exe` to `anything-else.exe` and see if that breaks?
